### ISSUE TITLE
list servers from smart inventory using awx cli

### ISSUE THREAD

**Issue by dipanjanmukherjee83** [ansible/awx#12613]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature Summary

It would be nice to have feature in awx cli to list servers from smart inventory.
Right now it is possible to list if we have custom made inventory but same cannot if inventory type is smart. Then same thing we can see using api and from gui as well.

for example here :
$ awx inventory list -f human --all |grep -i custom
65 custom_inventory

$ awx hosts list --inventory 65 -f human
id     name
====== ============
381712 srvxxx
381713 srvyyy

$ awx inventory list --kind smart -f human
id name
== ===========================================
24 Linux RH6/7/8 Acceptance Servers

$ awx hosts list --inventory 24 -f human
id name
== ====

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [X] CLI
- [ ] Other

**Comment by MalfuncEddie** [comment]
Maybe related but it would be nice if you could filter the inventory given an "ansible limit".

**Comment by dipanjanmukherjee83** [comment]
> Maybe related but it would be nice if you could filter the inventory given an "ansible limit".

Yes, we can do limit and even host patterns are working but the main issue is we cannot see the servers list which we want here.

**Comment by MalfuncEddie** [comment]
> > Maybe related but it would be nice if you could filter the inventory given an "ansible limit".
> 
> Yes, we can do limit and even host patterns are working but the main issue is we cannot see the servers list which we want here.

Sorry to go a bit of topic but are you sure?
if you do "awx hosts list --inventory 65 -f human" with a  filter it regexes the hosts and not apply the limit form cli?

a command like 
awx hosts list --inventory 65 -f human --filter " inventory group" 

does not return the hosts in a inventory group?
